src/server.c

```diff
@@ -4697,25 +4721,98 @@ sds genRedisInfoStringCommandStats(sds info, dict *commands) {
+   abd argc indicates the 2D array length
+
+   Another two parameter *out_all and *out)everything will be set value in this function based 
+   on the argv parameter.
+
+   The return value will be the dictionary for genRedisInfoString function
+ */
+dict *genInfoSectionDict(robj **argv, int argc, int *out_all, int *out_everything) {
+    char *defSections[] = {"server", "clients", "memory", "persistence", "stats", "replication", "cpu", "modules", "errorstats", "cluster", "keyspace", "latencystats"};
+    char *defSectionsSentinel[] = {"server", "clients", "cpu", "stats", "sentinel"};
+
+
+    dict *section_dict = dictCreate(&stringSetDictType);
+    if (server.sentinel_mode) {
+        if (argc == 1) {
+            addSectionsToDict(section_dict, defSectionsSentinel, sizeof(defSectionsSentinel)/sizeof(*defSectionsSentinel));
+        } else {
+            for (int i = 1; i < argc; i++) {
+                if (!strcasecmp(argv[i]->ptr,"default")) 
+                    addSectionsToDict(section_dict, defSectionsSentinel, sizeof(defSectionsSentinel)/sizeof(*defSectionsSentinel));
+                else if (!strcasecmp(argv[i]->ptr,"all")) 
+                    (*out_all) = 1;
+                else {
+                    sds section = sdsnew(argv[i]->ptr);
+                    sdstolower(section);
+                    dictAdd(section_dict,section,NULL);
+                }
+            }
+        }
+    } else {
+        if (argc == 1) {
+            if (!strcasecmp(argv[0]->ptr,"info") || !strcasecmp(argv[0]->ptr,"default")) 
+                addSectionsToDict(section_dict, defSections, sizeof(defSections)/sizeof(*defSections));
+            else if (!strcasecmp(argv[0]->ptr,"all")) 
+                    (*out_all) = 1;
+                else if (!strcasecmp(argv[0]->ptr,"everything")) 
+                    (*out_everything) = 1;
+            else {  
+                sds section = sdsnew(argv[0]->ptr);
+                sdstolower(section);
+                dictAdd(section_dict,section,NULL);
+                if (!strcasecmp(argv[0]->ptr,"modules")) {
+                    sds section = sdsnew("input-modules");
+                    dictAdd(section_dict,section,NULL);
+                }
+            }
+        } else {
+            for (int i = 1; i < argc; i++) {
+                if (!strcasecmp(argv[i]->ptr,"default")) 
+                    addSectionsToDict(section_dict, defSections, sizeof(defSections)/sizeof(*defSections));
+                else if (!strcasecmp(argv[i]->ptr,"all")) 
+                    (*out_all) = 1;
+                else if (!strcasecmp(argv[i]->ptr,"everything")) 
+                    (*out_everything) = 1;
+                else {
+                    sds section = sdsnew(argv[i]->ptr);
+                    sdstolower(section);
+                    dictAdd(section_dict,section,NULL);
+                    if (!strcasecmp(argv[i]->ptr,"modules")) {
+                        sds section = sdsnew("input-modules");
```

Comment:
Hi Oran, I want to explain why I add a fake part "input-modules" here. 
Below are the output results when user inputs 5 different cases.
![image](https://user-images.githubusercontent.com/51993843/148446007-13eb5585-44b8-4152-b012-c43a01b89041.png)

Reply:
i see the problem.
we actually have two sections for `modules`.
one is for the module list, printed by the genRedisInfoString and included by default, and the other is the module generated sections (not included by default, but is added by either `everything`, or `modules` args).

i think it will be clearer (and maybe also less code), if we change the default section list to include `module_list` instead of `modules`.
then the code in genRedisInfoString that generates the module list will depend on this section name (it'll still emit `# Modules` to be backwards compatible.
but when the user asks for "modules" we'll treat this as a request for the module generated sections.

Reply:
Update codes: add module_list in default section list